Classes/PKStateMachine.swift

```diff
@@ -0,0 +1,72 @@
+//
+//  PKStateMachine.swift
+//  Pods
+//
+//  Created by Gal Orlanczyk on 02/04/2017.
+//
+//
+
+import Foundation
+
+protocol IntRawRepresentable: RawRepresentable {
+    var rawValue: Int { get }
+}
+
+protocol StateProtocol: IntRawRepresentable, Hashable {}
+
+extension StateProtocol {
+    var hashValue: Int {
+        return rawValue
+    }
+}
+
+class BasicStateMachine<T: StateProtocol> {
```

Comment:
Please explain the motivation for this highly generic state machine. Do you plan to use it in other places?